publisher/publisher.go

```diff
@@ -306,10 +306,14 @@ func (pub *Impl) singleLogSubmit(
 		if canceled.Is(err) {
 			status = "canceled"
 		}
-		pub.metrics.submissionLatency.With(prometheus.Labels{
+		labels := prometheus.Labels{
 			"log":    ctLog.uri,
 			"status": status,
-		}).Observe(took)
+		}
+		if rspError, ok := err.(*ctClient.RspError); ok && rspError.StatusCode != 0 {
+			labels["httpStatus"] = fmt.Sprintf("%d", rspError.StatusCode)
```

Comment:
I think this approach is causing real Travis failures:
> panic: inconsistent label cardinality

The number of keys in the `prometheus.Labels` map needs to be consistent even if the err is not a `*ctClient.RspError`

Reply:
Totally forgot about this requirement. Instead of just using `0` when everything is good what do you think about just appending the status code to the `status` field if we have a `ctClient.RspError` (i.e. `error [666]` or something)?

Reply:
Hmm. I feel like that will be harder to process in Prometheus queries. I think using `0` (or `http.StatusOK`) for the everything is good case would be easier to filter out with promql.